server/id/id_test.go

```diff
@@ -4,44 +4,72 @@ package id
 
 import (
 	"context"
-	"path"
-	"strconv"
+	"sync"
 	"testing"
 	"time"
 
 	"github.com/CeresDB/ceresmeta/server/etcdutil"
-	"github.com/CeresDB/ceresmeta/server/storage"
-	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	clientv3 "go.etcd.io/etcd/client/v3"
 	"go.etcd.io/etcd/server/v3/embed"
 )
 
-const defaultRequestTimeout = time.Second * 10
+const (
+	defaultRequestTimeout = time.Second * 30
+	defaultStep           = 100
+	defaultRootPath       = "/ceresmeta"
+	defaultAllocIDKey     = "/id"
+)
 
-// TODO: need a concurrent test case
-func TestAlloc(t *testing.T) {
+func newTestKV(t *testing.T) clientv3.KV {
 	re := require.New(t)
 	cfg := etcdutil.NewTestSingleConfig()
 	etcd, err := embed.StartEtcd(cfg)
-	assert.NoError(t, err)
-
+	re.NoError(err)
 	<-etcd.Server.ReadyNotify()
 
 	endpoint := cfg.LCUrls[0].String()
 	client, err := clientv3.New(clientv3.Config{
 		Endpoints: []string{endpoint},
 	})
-	assert.NoError(t, err)
-	rootPath := path.Join("/ceresmeta", strconv.FormatUint(100, 10))
-	kv := storage.NewEtcdKV(client, rootPath)
+	re.NoError(err)
+
+	return client
+}
+
+func TestAllocMultiThread(t *testing.T) {
```

Comment:
- A better name `TestMultipleAllocBasedOnKV`
- No need to allocate id parallelly.